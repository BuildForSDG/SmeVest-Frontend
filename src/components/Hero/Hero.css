.Hero {
  width: 100%;
  height: auto;
  background: url('../../assets/images/hero.webp');
  background-size: 65% 80%;
  background-repeat: no-repeat;
  background-position: top right -10%;
  margin: 3rem auto 1rem;
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding: 2rem 0 20rem;
  font-family: 'Karla', sans-serif;
}

.Hero .GetStarted {
  width: 40%;
  margin-left: 0;
  margin-top: 0rem;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.Hero h1 {
  font-family: 'Karla', sans-serif;
  font-weight: 700;
  font-size: 2.8rem;
  text-align: center;
  color: #212226;
  margin-bottom: 1rem;
}

.Hero .SignUp {
  text-decoration: none;
  font-size: 0.95rem;
  padding: 0.4rem 1.5rem;
  transition: all 300ms linear;
  background: #4ad6e9;
  border-radius: 1rem;
  color: #fff;
  box-shadow: 0 2px 1px #1fc2d8, 0 -1px 1px #1fc2d8;
  border: 1px solid #4ad6e9;
}

.Hero .SignUp:hover {
  background: transparent;
  color: #212226;
  box-shadow: 0 0px 0px rgba(0, 0, 0, 0.7), 0px 0px #fff;
  box-sizing: border-box;
}

@media screen and (max-width: 768px) {
  .Hero {
    background-size: contain;
    background-repeat: no-repeat;
    background-position: center center;
    margin: 2rem 0 0rem;
    flex-direction: row;
    align-items: flex-start;
    padding: 0rem 0 23rem;
  }

  .Hero .GetStarted {
    width: 100%;
    margin: auto;
    padding: 1rem 0;
  }

  .Hero h1 {
    font-weight: 500;
    font-size: 2rem;
    line-height: 1.2;
    margin-bottom: 1rem;
  }
}
