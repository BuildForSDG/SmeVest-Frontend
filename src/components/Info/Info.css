.Info {
  width: 100%;
  min-height: 40vh;
  height: auto;
  background: #eee;
  padding: 10rem 0 5rem;
  position: relative;
  font-family: 'Karla', sans-serif;
}

.Info .Icons {
  position: absolute;
  border-radius: 1rem;
  background: #fff;
  width: 400px;
  right: 7rem;
  top: -3.8rem;
  display: flex;
  justify-content: space-evenly;
  align-items: center;
  text-align: center;
  padding: 1rem 0 0.5rem;
  box-shadow: 0 0 0.5rem #ddd;
}

.Info .Icon > .Icons-svg {
  animation-name: zoom;
  animation-duration: 2s;
  animation-fill-mode: both;
  animation-iteration-count: infinite;
  animation-direction: alternate-reverse;
}

.Info .Icon .Icons-Text {
  margin: 0.3rem 0;
  font-size: 1.1rem;
  font-family: 'Karla', sans-serif;
  font-weight: 300;
  letter-spacing: 0.1rem;
  text-transform: uppercase;
}

.Info .Icon:not(:last-child) {
  border-right: 1px solid #eee;
  padding-right: 2rem;
}

.Detail {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
}

.Detail .fast,
.Detail .secure,
.Detail .connect {
  padding: 1rem 1rem 6rem;
  background: #fff;
  border-radius: 0.35rem;
  height: 80px;
  width: 300px;
  margin: -2rem 0 4rem;
  transition: all 500ms ease;
}

.Detail .fast {
  box-shadow: 0 -3px 0 #d4475a;
  margin-left: -10rem;
}

.Detail .secure {
  box-shadow: 0 -3px 0 #47d486;
}

.Detail .connect {
  box-shadow: 0 -3px 0 #4789d4;
  margin-left: -10rem;
}

.Detail-Icon {
  margin-right: 0.3rem;
}

.fast .Detail-Icon {
  color: #d4475a;
}

.secure .Detail-Icon {
  color: #47d486;
}

.connect .Detail-Icon {
  color: #4789d4;
}

.Detail-Heading,
.Detail-Text {
  margin: 0;
  font-family: 'Baloo 2', sans-serif;
}

.Detail-Heading {
  font-weight: 400;
  text-transform: uppercase;
  font-size: 1.2rem;
  letter-spacing: 0.1rem;
  text-align: center;
  border-bottom: 1px solid #eee;
  padding-bottom: 0.4rem;
  margin-bottom: 0.4rem;
  display: flex;
  justify-content: center;
  align-items: center;
}

.Detail-Text {
  color: #b6b0b0;
  font-family: Karla, sans-serif;
  margin: 0.3rem 0;
}

.Detail .fast:hover {
  box-shadow: 0 -3px 0 #fff;
  background: #d4475a;
  color: #fff;
  margin-left: 0;
}

.Detail .secure:hover {
  box-shadow: 0 -3px 0 #fff;
  background: #47d486;
  color: #fff;
  margin-left: -10rem;
}

.Detail .connect:hover {
  box-shadow: 0 -3px 0 #fff;
  background: #4789d4;
  color: #fff;
  margin-left: 0;
}

.Detail .fast:hover .Detail-Icon,
.Detail .fast:hover .Detail-Text,
.Detail .secure:hover .Detail-Icon,
.Detail .secure:hover .Detail-Text,
.Detail .connect:hover .Detail-Icon,
.Detail .connect:hover .Detail-Text {
  color: #fff;
}

.Image-responsive {
  border-radius: 1rem;
  overflow: hidden;
  box-shadow: 2px 2px 0.3rem rgba(0, 0, 0, 0.2), -2px 0 0.3rem rgba(0, 0, 0, 0.2);
  width: 100%;
  height: auto;
  max-width: 100%;
  display: block;
}

/* Info Section 2 */
.Works {
  width: 100%;
  padding: 8rem 0 10rem;
  box-sizing: border-box;
  position: relative;
  font-family: 'Karla', sans-serif;
}

.LeftImage {
  width: 300px;
  height: 300px;
  border-radius: 50%;
  overflow: hidden;
  box-sizing: border-box;
  background: url('../../assets/images/info2.webp');
  background-size: contain;
  background-position: center right;
  box-shadow: 0 0 0.5rem rgba(0, 0, 0, 0.4);
  animation-name: spin;
  animation-duration: 5s;
  animation-fill-mode: both;
  animation-iteration-count: 3;
  animation-direction: alternate-reverse;
  animation-timing-function: ease-in-out;
  align-self: center;
}

.Work-Detail-Heading {
  font-size: 2rem;
  color: #aaa;
}

.Work-Detail-Text {
  display: flex;
  align-items: center;
  font-size: 1.2rem;
}

.Work-Detail-Icon {
  margin-right: 0.5rem;
  border-right: 2px solid #47d486;
  padding-right: 0.3rem;
}

/* Ready Section Footer Friend */
.Ready {
  width: 400px;
  background: #fff;
  color: #212226;
  position: absolute;
  padding: 1rem;
  box-sizing: border-box;
  bottom: -4.2rem;
  left: 50%;
  transform: translateX(-50%);
  border-radius: 0.5rem;
  box-shadow: 0 0 0.1rem rgba(0, 0, 0, 0.4);
  text-align: center;
  font-family: 'Karla', sans-serif;
}

.Ready-Text {
  font-size: 1.2rem;
}

.Ready-Btn {
  text-decoration: none;
  font-size: 0.95rem;
  padding: 0.4rem 1.5rem;
  transition: all 300ms linear;
  background: #4ad6e9;
  border-radius: 1rem;
  color: #fff;
  box-shadow: 0 2px 1px #1fc2d8, 0 -1px 1px #1fc2d8;
  border: 1px solid #4dd7e9;
  margin: 0.5rem 0;
  display: inline-block;
}

.Ready-Btn:hover {
  background: transparent;
  color: #212226;
  box-shadow: 0 0 0 rgba(0, 0, 0, 0.7), 0 0 #fff;
  box-sizing: border-box;
  text-decoration: none;
}

@media screen and (max-width: 1200px) {
  .Info .Icons {
    position: absolute;
    width: 350px;
    right: 50%;
    transform: translateX(50%);
  }

  .Info .Icon .Icons-Text {
    font-size: 0.95rem;
  }

  .Info .Detail {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    align-items: center;
  }

  .Info .Detail .fast,
  .Info .Detail .connect {
    margin-left: 0;
  }

  .Info .Detail .secure:hover {
    margin-left: 0;
  }

  .Works {
    padding: 5rem 0;
  }

  .LeftImage {
    order: 2;
    justify-self: center;
    background-position: center center;
    text-align: center;
    margin: 0 auto 2rem;
  }

  .Work-Detail-Heading {
    font-size: 1.6rem;
    text-align: center;
  }

  .Work-Detail-Text {
    margin: 2rem 0;
    align-items: flex-start;
    font-size: 1rem;
    text-align: left;
  }

  .Ready {
    width: 350px;
  }
}

@media screen and (max-width: 320px) {
  .Info .Icons {
    width: 300px;
  }

  .Info .Icon .Icons-Text {
    font-size: 0.9rem;
  }

  .Works {
    padding: 3rem 0 8rem;
  }

  .LeftImage {
    width: 280px;
    height: 280px;
    margin: 0 auto 2rem;
    background-position: center center;
  }

  .Work-Detail-Heading {
    margin-bottom: 0;
  }

  .Ready {
    width: 300px;
    bottom: -3.6rem;
  }

  .Ready-Text {
    font-size: 1rem;
  }
}

@keyframes zoom {
  from {
    transform: scale(0.9);
  }

  to {
    transform: scale(1.04);
  }
}

@keyframes spin {
  from {
    transform: rotate(0);
  }

  to {
    transform: rotate(360deg);
  }
}
