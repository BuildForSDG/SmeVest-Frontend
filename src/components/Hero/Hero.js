import React from 'react';
import { Link } from 'react-router-dom';
import './Hero.css';
import { Container, Row } from 'react-bootstrap';

const Hero = ({ currentUser }) => {
  return (
    <Container>
      <div className="Hero">
        <Row className="GetStarted">
          <h1>Welcome To Your Gateway Of Business Investment</h1>
          {!currentUser && (
            <Link to="/signup" className="SignUp">
              Get Started For Free
            </Link>
          )}
        </Row>
      </div>
    </Container>
  );
};

export default Hero;
