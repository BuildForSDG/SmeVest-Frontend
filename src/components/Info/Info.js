import React from 'react';
import { Link } from 'react-router-dom';
import {
  OfflineBolt,
  Security,
  SupervisorAccount,
  ArrowRightAlt,
  AllInclusive,
  VerifiedUser,
  Message,
} from '@material-ui/icons';
import './Info.css';
import { Container, Row, Col } from 'react-bootstrap';
import infoImage from '../../assets/images/info-image.webp';

const loggedIn = false;
const Info = () => (
  <>
    <div className="Info">
      <Container>
        <section className="Icons">
          <div className="Icon">
            <OfflineBolt className="Icons-svg" style={{ fontSize: 60, color: '#d4475a' }} />
            <p className="Icons-Text">Fast</p>
          </div>
          <div className="Icon">
            <Security className="Icons-svg" style={{ fontSize: 60, color: '#47d486' }} />
            <p className="Icons-Text">Secure</p>
          </div>
          <div className="Icon">
            <SupervisorAccount className="Icons-svg" style={{ fontSize: 60, color: '#4789d4' }} />
            <p className="Icons-Text">Connect</p>
          </div>
        </section>
        <section className="Details">
          <Row>
            <Col md>
              <div className="Detail">
                <div className="fast">
                  <h2 className="Detail-Heading">
                    <OfflineBolt className="Detail-Icon" />
                    Fast Connection
                  </h2>
                  <p className="Detail-Text">A reliable system that offers fast connection</p>
                </div>
                <div className="secure">
                  <h2 className="Detail-Heading">
                    <Security className="Detail-Icon" />
                    Secured Connection
                  </h2>
                  <p className="Detail-Text">Your connection is secured with this system</p>
                </div>
                <div className="connect">
                  <h2 className="Detail-Heading">
                    <SupervisorAccount className="Detail-Icon" />
                    SME/Investor
                  </h2>
                  <p className="Detail-Text">The right connection is waiting for you</p>
                </div>
              </div>
            </Col>
            <Col sm>
              <div className="RightImage">
                <img src={infoImage} alt="Info" className="Image-responsive" />
              </div>
            </Col>
          </Row>
        </section>
      </Container>
    </div>

    <div className="Works">
      <Container>
        <Row>
          <Col sm>
            <div className="LeftImage"></div>
          </Col>
          <Col md="auto">
            <div className="Work-Detail">
              <h2 className="Work-Detail-Heading">How The System Works</h2>
              <p className="Work-Detail-Text">
                <ArrowRightAlt className="Work-Detail-Icon" style={{ fontSize: 40, color: '#47d486' }} /> Automatically
                suggest SMEs/Investors matching Your Category
              </p>
              <p className="Work-Detail-Text">
                <AllInclusive className="Work-Detail-Icon" style={{ fontSize: 40, color: '#4789d4' }} /> Connects you to
                the SME/Investor
              </p>
              <p className="Work-Detail-Text">
                <Message className="Work-Detail-Icon" style={{ fontSize: 40, color: '#4789d4' }} /> On Successful
                connection, a private messaging system is enabled for both parties
              </p>
              <p className="Work-Detail-Text">
                <VerifiedUser className="Work-Detail-Icon" style={{ fontSize: 40, color: '#47d486' }} />
                <span
                  style={{
                    color: '#d4475a',
                    display: 'inline-block',
                    marginRight: 4,
                  }}
                >
                  Note:
                </span>
                Only connect to verified SME/Investor
              </p>
            </div>
          </Col>
        </Row>
      </Container>

      {!loggedIn && (
        <div className="Ready">
          <h2 className="Ready-Text">Ready To Make That Connection?</h2>
          <Link to="/signup" className="Ready-Btn">
            Get Started For Free
          </Link>
        </div>
      )}
    </div>
  </>
);

export default Info;
