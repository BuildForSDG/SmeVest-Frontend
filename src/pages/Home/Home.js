import React from 'react';
import './Home.css';
import Hero from '../../components/Hero/Hero';
import Info from '../../components/Info/Info';

const Home = () => (
  <>
    <Hero />
    <Info />
  </>
);

export default Home;
