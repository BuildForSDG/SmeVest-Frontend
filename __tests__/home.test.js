import React from 'react';
import { shallow } from 'enzyme';
import Home from '../src/pages/Home/Home.js';
import Hero from '../src/components/Hero/Hero.js';
import Info from '../src/components/Info/Info.js';

describe('<Home /> Component', () => {
  const wrapper = shallow(<Home />);
  test('should render home component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should contain hero component and info component', () => {
    expect(wrapper.containsAllMatchingElements([<Hero key="1" />, <Info key="2" />])).toBe(true);
  });
});
