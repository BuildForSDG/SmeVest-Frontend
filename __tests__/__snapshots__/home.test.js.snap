// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> Component should render home component 1`] = `ShallowWrapper {}`;
